### PR TITLE
Add preStop hook, update plans, fix custom images

### PR DESCRIPTION
**Pull request by rogeruiz**
This triggers a manual failover whenever a Master Redis server
shutsdown.

pairing with @linuxbozo, @jmcarp

### CODE REVIEW

**Comment by rogeruiz** [comment]
So we've got another race condition here. I managed to get 4 iterations of tests to pass with Redis + Sentinel, but I'm still seeing some issues when the Replica that's been elected a Primary hasn't synchronized and can't seem to get their config rewritten because of a permission denied error.

```
1:S 11 Aug 04:48:36.239 * MASTER <-> SLAVE sync started
1:M 11 Aug 04:48:40.736 * Discarding previously cached master state.
1:M 11 Aug 04:48:40.736 * MASTER MODE enabled (user request from 'id=4 addr=local-ip:56530 fd=5 name=sentinel-b5bbc6db-cmd age=101 idle=0 flags=x db=0 sub=0 psub=0 multi=3 qbuf=0 qbuf-free=32768 obl=36 oll=0 omem=0 events=r cmd=exec')
1:M 11 Aug 04:48:40.737 # CONFIG REWRITE failed: Permission denied
1:M 11 Aug 04:48:41.894 * Slave local-ip:6379 asks for synchronization
1:M 11 Aug 04:48:41.894 * Full resync requested by slave local-ip:6379
```

Looking at the pods for Redis Server it looks like like `redis-server` commands runs as `redis` instead of `root`.

File: catalogData/redis32/service.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
     "id": "9f54be76-722b-11e7-ad8f-6fd2bc5f79fe",
     "name": "redis32",
-    "description": "An open source in-memory data structure store.",
+    "description": "An open source in-memory data structure store with Redis Sentinel for high-availability.",
     "metadata":{
-      "displayName": "Redis 3.2",
+      "displayName": "Redis 3.2.10",
```
Comment (LinuxBozo) [inline catalogData/redis32/service.json]: I would probably leave this as 3.2 so we can upgrade minor versions for security, etc.

File: catalogData/redis32/service.json [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
       "providerDisplayName": "18F",
-      "documentationUrl": "https://cloud.gov/docs/services/redis/",
+      "documentationUrl": "https://cloud.gov/docs/services/redis/#3.2.10",
```
Comment (LinuxBozo) [inline catalogData/redis32/service.json]: Same here, use 3.2

File: catalogData/redis32/standard-ha/plan.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "id": "7238138e-722b-11e7-9e57-c30dd7276550",
+  "name": "standard-ha",
+  "description": "Redis 3.2.10 with Redis Sentinel and persistent storage.",
```
Comment (LinuxBozo) [inline catalogData/redis32/standard-ha/plan.json]: Again, would just use 3.2

**Comment by rogeruiz** [comment]
@cnelson awesome. should this be happening as part of the `pre-stop` for servers, or do you think it's fine for it to just happen on the start of sentinels? Because of the IP reuse, we were seeing weird `slave` properties because at one point they were servers, am I understanding the problem right?

**Review by sharms — APPROVED** [review]

**Pull request merged**